test/inheritAttributes.test.js

```diff
@@ -490,3 +494,869 @@ QUnit.test('end to end - alternate BaseURLs', function(assert) {
+      <BaseURL>https://www.example.com/base/</BaseURL>
+      <Period duration= "PT0H4M40.414S" >
+        <AdaptationSet mimeType= "video/mp4"  >
+          <Role value= "main" ></Role>
+          <SegmentBase indexRange= "1212"  indexRangeExact= "true" >
+           <Initialization range= "0-8888"  />
+          </SegmentBase>
+          <Representation
+            mimeType= "video/mp6"
+            bandwidth= "5000000"
+            codecs= "avc1.64001e"
+            height= "404"
+            id= "test"
+            width= "720" >
+            <SegmentBase></SegmentBase>
+          </Representation>
+          <Representation
+            height= "545" >
+            <SegmentBase indexRangeExact= "false" >
+            </SegmentBase>
+          </Representation>
+        </AdaptationSet>
+        <AdaptationSet mimeType= "text/vtt"  lang= "en" >
+          <Representation bandwidth= "256"  id= "en" >
+            <BaseURL>https://example.com/en.vtt</BaseURL>
+          </Representation>
+        </AdaptationSet>
+      </Period>
+    </MPD>
+  `
+  ));
+
+  const expected = [{
+    attributes: {
+      bandwidth: '5000000',
+      baseUrl: 'https://www.example.com/base/',
+      duration: 'PT0H4M40.414S',
+      codecs: 'avc1.64001e',
+      height: '404',
+      id: 'test',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'video/mp6',
+      periodIndex: 0,
+      role: {
+        value: 'main'
+      },
+      sourceDuration: 30,
+      width: '720'
+    },
+    segmentInfo: {
+      base: {
+        indexRange: '1212',
+        indexRangeExact: 'true',
+        initialization: {
+          range: '0-8888'
+
+        }
+      },
+      list: undefined,
+      template: undefined,
+      timeline: undefined
+    }
+  }, {
+    attributes: {
+      baseUrl: 'https://www.example.com/base/',
+      mediaPresentationDuration: 'PT30S',
+      duration: 'PT0H4M40.414S',
+      mimeType: 'video/mp4',
+      periodIndex: 0,
+      height: '545',
+      role: {
+        value: 'main'
+      },
+      sourceDuration: 30
+    },
+    segmentInfo: {
+      base: {
+        indexRange: '1212',
+        indexRangeExact: 'false',
+        initialization: {
+          range: '0-8888'
+        }
+      },
+      list: undefined,
+      template: undefined,
+      timeline: undefined
+    }
+  }, {
+    attributes: {
+      bandwidth: '256',
+      baseUrl: 'https://example.com/en.vtt',
+      duration: 'PT0H4M40.414S',
+      id: 'en',
+      lang: 'en',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'text/vtt',
+      periodIndex: 0,
+      role: {},
+      sourceDuration: 30
+    },
+    segmentInfo: {
+      base: undefined,
+      list: undefined,
+      template: undefined,
+      timeline: undefined
+    }
+  }];
+
+  assert.equal(actual.length, 3);
+  assert.deepEqual(actual, expected);
+});
+
+QUnit.test(' Test for checking presence of atmost one segment at each level', function(assert) {
```

Comment:
This description is a bit misleading maybe something along the lines of "uses only one set of segment information when multiple are present" and then add a code comment saying that although the spec states that at each level at most one shall be present, we still handle the case to prevent throwing errors